### PR TITLE
Expose Transfer endpoints

### PR DESCRIPTION
**Pull request by OdunlamiZO**

### CODE REVIEW

File: model/transfer.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		BeneficiaryID       *string              `json:"beneficiary_id,omitempty"`
+		Destination         *TransferDestination `json:"destination,omitempty"`
+		Note                *string              `json:"note,omitempty"`
+		Reason              string               `json:"reason"`
+	}
+
+	// TransferResponse schema for transfer response
+	TransferResponse struct {
+		ID uuid.UUID `json:"id"`
+		InitiateTransferRequest
```
Comment (wharleyinc) [inline model/transfer.go:45]: you can add a json tag

Comment (OdunlamiZO) [reply model/transfer.go:45]: It's returned like that from external api

File: model/transfer.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		Note            *string        `json:"note"`
+		Reason          string         `json:"reason"`
+		CreatedAt       time.Time      `json:"created_at"`
+		CompletedAt     sql.NullTime   `json:"completed_at"`
+		UpdatedAt       sql.NullTime   `json:"updated_at"`
+		BatchDate       time.Time      `json:"batch_date"`
+		Status          string         `json:"status"`
+		Reference       string         `json:"reference"`
+		CancelReason    *string        `json:"cancel_reason"`
+		TransactionType string         `json:"type"`
```
Comment (itebababatunde) [inline model/transfer.go:77]: there's a mismatch btw the json tag and field name

Comment (OdunlamiZO) [reply model/transfer.go:77]: `type` is being used as the query param on external api

**Review by odetolakehinde — APPROVED** [review]

File: model/transfer.go (hunk 2 above)
Comment (itebababatunde) [reply model/transfer.go:77]: noted

**Review by Afunnilo — APPROVED** [review]

**Review by arkmadj — APPROVED** [review]

**Pull request merged**